pallet-chainlink-feed/src/lib.rs

```diff
@@ -711,7 +733,14 @@ pub mod pallet {
 					let updated_at = frame_system::Pallet::<T>::block_number();
 					round.updated_at = Some(updated_at);
 					round.answered_in_round = Some(round_id);
-					Rounds::<T>::insert(feed_id, round_id, round);
+					Rounds::<T>::insert(feed_id, round_id, round.clone());
+
+					// # Safty
+					//
+					// this will never fail, because we just filled every fields in
+					// last lines
+					let round_on_answer =
+						round.try_into().map_err(|_| <Error<T>>::RoundConversion)?;
```

Comment:
Since we constructed the `round` above, we should also create a `RoundData` using the fields directly, so that we can get rid of the `RoundConversion`, we need to clone the round anyways.

Reply:
Or a `RoundData` with `impl Into<Round> for RoundData`

Clone every field take too many lines